### DISCUSSION TITLE
Can't setup fabric on Mac Os

### DISCUSSION THREAD

**Discussion by ApiAGIIntelligent** [danielmiessler/Fabric#936] [Q&A]
i used the new method of downloading it with go i'm using an apple silicon chip and when i say "fabric --setup" it says "fabric" is not a command. I already tried reopening terminal many times nothing is working.

**Comment by alexxmugg** [comment]
Go Module Commands

1. Check for go.mod
ls go.mod

2. Create go.mod
go mod init "module_name"

3. Add dependency
go get github.com/username/repo

4. Import in code
import "github.com/username/repo"

5. Update go.mod
go mod tidy

6. Update dependencies
go get -u

7. List dependencies
go list -m all

**Reply by ApiAGIIntelligent** [reply]
this isn't helping.

**Reply by ApiAGIIntelligent** [reply]
zsh : cannot find command fabric.

**Reply by ApiAGIIntelligent** [reply]
i don't know how to add the module to go

**Reply by alexxmugg** [reply]
The installation link is in the README, https://go.dev/doc/install, 

Then execute the commands from above

**Reply by ApiAGIIntelligent** [reply]
i downloaded go now what do i do?

**Reply by ApiAGIIntelligent** [reply]
> The installation link is in the README, https://go.dev/doc/install,
> 
> Then execute the commands from above

zsh: command not found: Is

**Reply by alexxmugg** [reply]
1. Ensure Go is installed
   go version

Configuration
1. Open the .zshrc file
   vim ~/.zshrc

2. Add the following lines at the end of the file:
   export GOROOT=/usr/local/go
   export GOPATH=$HOME/go
   export PATH=$GOPATH/bin:$GOROOT/bin:$PATH
   alias fabric=$HOME/go/bin/fabric

3. Save and exit vim
   :wq

4. Reload the .zshrc file
   source ~/.zshrc

**Reply by ApiAGIIntelligent** [reply]
> The installation link is in the README, https://go.dev/doc/install,
> 
> Then execute the commands from above

I made a module then i added the dependency but on number 4 what do i do i don't understand.

**Reply by ApiAGIIntelligent** [reply]
Swap file "~/.zshrc.swp" already exists!
[O]pen Read-Only, (E)dit anyway, (R)ecover, (D)elete it, (Q)uit, (A)bort:

**Reply by alexxmugg** [reply]
Choose Recover, you have 2 files open a the same time

**Reply by ApiAGIIntelligent** [reply]
how do i exit the file

**Reply by ApiAGIIntelligent** [reply]
at the topwhere it says export GOROOT = /usr/local/go do i put my username in for "usr"

**Reply by alexxmugg** [reply]
> at the topwhere it says export GOROOT = /usr/local/go do i put my username in for "usr"

no

**Reply by alexxmugg** [reply]
> how do i exit the file



> 1. Ensure Go is installed
>    go version
> 
> Configuration
> 
> 1. Open the .zshrc file
>    vim ~/.zshrc
> 2. Add the following lines at the end of the file:
>    export GOROOT=/usr/local/go
>    export GOPATH=$HOME/go
>    export PATH=$GOPATH/bin:$GOROOT/bin:$PATH
>    alias fabric=$HOME/go/bin/fabric
> 3. Save and exit vim
>    :wq
> 4. Reload the .zshrc file
>    source ~/.zshrc

**Reply by ApiAGIIntelligent** [reply]
do i delete whats already in the zshrc file

**Reply by ApiAGIIntelligent** [reply]
when i type :wq it doesn't exit

**Reply by alexxmugg** [reply]
try "esc" then ":wq"

**Reply by ApiAGIIntelligent** [reply]
do i put it on the first ~

**Reply by ApiAGIIntelligent** [reply]
alr i exited it

**Reply by ApiAGIIntelligent** [reply]
now what do i do?

**Comment by ApiAGIIntelligent** [comment]
and there is no "end" of the file i just keep scrolling and scrolling

**Reply by alexxmugg** [reply]
the "end" is the last line of code...

**Reply by ApiAGIIntelligent** [reply]
all i saw was #pipx something something then 
something to do with .local/bin then a whole bunch of lines with
~
~
~

**Reply by alexxmugg** [reply]
don't care about the "~" it's normal, the end of the file is the last line of code

**Reply by ApiAGIIntelligent** [reply]
i did the fabric --setup thing and it's asking for an open ai api key
